code/books4u/api.py

```diff
@@ -16,3 +17,143 @@ def get_all_books(request):
+            account_password = make_password(request.POST.get('password'))
+            account_e_mail = request.POST.get('e_mail')
+            account_gender = request.POST.get('gender')
+        except ValueError:
+            HttpResponse(json.dumps({'status': 'failure_get'}))
+        try:
+            new_user = User(name=account_name, password=account_password, e_mail=account_e_mail, gender=account_gender)
+            new_user.save()
+        except(TypeError, ValueError):
+            HttpResponse(json.dumps({'status': 'failure_save'}))
+        return HttpResponse(json.dumps({'status': 'success'}))
+    else:
+        return HttpResponse(json.dumps({'status': 'failure_method'}))
+
+""" used by users to add a new book
+    visibility of the new book is false, representing not officailly added
+"""
+def add_book(request):
+
+    if request.method == "POST":
+        try:
+            ISBN_ = request.POST.get('ISBN')
+            name_ = request.POST.get('name')
+            publish_date_ = request.POST.get('publish_date','')
+            edition_ = request.POST.get('edition','')
+            category_ = request.POST.get('category')
+            author_ = request.POST.get('author')
+        except requests.exceptions.RequestException:
+            return HttpResponse(json.dumps({'status': 'missing_required_fields'}))
+
+        exist = False
+        #check if the input category exists
+        categories = BookCategory.objects.all()
+        for c in categories:
+            if c.name == category_:
+                exist = True
+                category_o = c
+                break
+        if not exist:
+            category_o = BookCategory(name=category_)
+
+        exist = False
+        #check if the input author exists
+        authors = Author.objects.all()                        
+        for a in Author:
+            if a.name == author_:
+                exist = True
+                author_o = a
+                break
+        if not exist:
+            author_o = Author(name=author_, summary='')
+        
+        new_book = Book(ISBN=ISBN_, name=name_, publish_date=publish_date_, edition=edition_, category_o=category_, author_o=author_)
+        new_book.save()
+    
+    else:
+        return HttpResponse(json.dumps({'status': 'failure_method'}))
+
+""" call this method to check if the moderator should be notified that there is a pending new book
+    return a list of books needed to be verifid
+"""
+def get_pending_books():
+
+    pending_books = []
+    books = Book.objects.all()
+    for book in books:
+        if book.visibility == False:
+            pending_books.append(book)
+
+    return pending_books
+
+""" request contains two values:
+    a bool indicating permmited or not and a str indicating the ISBN of a pending book
+"""
+def commit_book(request):
+
+    if request.method == "POST":
+        try:
+            ISBN_ = request.POST.get('ISBN')
+            permmited = request.POST.get('permit')
```

Comment:
For this method I assumed two variables passed by the request, an 'ISBN' indicates the book which is currently been verified, and a boolean 'permmit' indicates wether moderator permitted or not. But it may be better to seperate permitted and not permitted into two methods, which can be called by two different buttons maybe, so there is no non-field variable passed in.